client/session.go

```diff
@@ -3313,7 +3321,9 @@ func (r *bulkBlocksResult) addBlockFromPeer(id ident.ID, peer topology.Host, blo
 		result.Close()
 
 		result = r.blockOpts.DatabaseBlockPool().Get()
-		result.Reset(start, encoder.Discard())
+		blockSize := resultReader.End.Sub(start)
```

Comment:
Do you need to override `blockSize` here?  Using the variable passed to the method might be more accurate than this calculation here, since its passed directly from the retention options rather than by calculating it.